### PR TITLE
Add new lint for space in Headers

### PR DESCRIPTION
**Pull request by 0xRampey**
Addresses Issue https://github.com/ethereum/eipw/issues/40

Also have to add docs for the new lint. Is there a cargo command to generate the required `html` file?

### CODE REVIEW

File: eipw-lint/tests/lint_markdown_headings_space.rs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    assert_eq!(
+        reports,
+        r#"error[markdown-headings-space]: Space missing in header
+  |
+5 | ##B#an#ana
+  |  ^ space required here
+  |
+"#
+    );
+}
```
Comment (SamWilsn) [inline eipw-lint/tests/lint_markdown_headings_space.rs:71]: Would you mind adding a test or two for cases these examples:

```markdown
*Hello*#world
```

```markdown
`#world`
```

**Review by SamWilsn — CHANGES_REQUESTED** [review]
Hey! Thanks so much for the contribution! I completely missed this pull request.

Just these two test cases, and I think it'll be good to go.

**Comment by SamWilsn** [comment]
> Is there a cargo command to generate the required html file?

Nope, they're just done by hand

File: eipw-lint/tests/lint_markdown_headings_space.rs (hunk 1 above)
Comment (0xRampey) [reply eipw-lint/tests/lint_markdown_headings_space.rs:71]: Good catch! Looks like in the case of `*Hello*#world`, `*Hello*` and `#world` are considered as separate text nodes by the linter and `#world` is being flagged. 

Will take a look at parsing using `paragraph` nodes instead.

**Pull request merged**